File: judgels-backends/jophiel/jophiel-api/src/main/java/judgels/jophiel/api/role/Role.java
package judgels.jophiel.api.role;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

public enum Role {
    @JsonProperty ADMIN("admin"),
    @JsonProperty SUPERADMIN("superadmin"),
    @JsonProperty USER("user");

    private final String text;

    Role(String text) {
        this.text = text;
    }

    @JsonValue
    public String toString() {
        return this.text;
    }
}

File: judgels-frontends/raphael/src/modules/api/jophiel/role.js
export interface UserRole {
  jophiel: JophielRole;
  sandalphon?: string;
  uriel?: string;
  jerahmeel?: string;
}

export enum JophielRole {
  Guest = 'GUEST',
  User = 'USER',
  Admin = 'ADMIN',
  Superadmin = 'SUPERADMIN',
}
